### PR TITLE
Move communication to SignalR

### PR DESCRIPTION
**Pull request by krzysiekprzekwas**
To achive that i had to:

- Change framework from 2.0 to 2.1
- Restructurized project into single project
- Turn off logger
- Introduce GameHub
- Rewrite client reciving logic

### CODE REVIEW

File: GameServer/wwwroot/js/connection.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            .withUrl('/game')
+            .build();
+
+        this.connection.onclose = function () {
+            console.log('connecition closed');
+        };
+
+        // Create a function that the hub can call to broadcast messages.
+        this.connection.on('updateGameState', function (gameState) {
+            console.log("UpdatedState");
```
Comment (tomejzen) [inline GameServer/wwwroot/js/connection.js]: This will spam a lot.

File: GameServer/wwwroot/js/connection.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+            world.onGameStateReceived(gameState);
+        });
+
+        this.connection.on('connectConfirmation', function (response) {
+            logger.info(`Player connected #${response.playerId}`);
```
Comment (tomejzen) [inline GameServer/wwwroot/js/connection.js]: We decided not to use logger anymore. You can delete it or I will do it during my front changes.

File: GameServer/wwwroot/js/connection.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+            for (setting in response.config) {
+
+                const value = response.config[setting.toUpperCase()];
+                config[setting] = value;
```
Comment (tomejzen) [inline GameServer/wwwroot/js/connection.js]: Are You sure we can access this config variable from anywhere else?

File: GameServer/wwwroot/js/connection.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            world.playerId = response.playerId;
+        });
+
+        // Transport fallback functionality is now built into start.
+        this.connection.start()
+            .then(function () {
+                console.log('connection started');
+                setTimeout(connector.onOpen(), 1000);
+
+                this.messagesReceivedCount = 0;
```
Comment (tomejzen) [inline GameServer/wwwroot/js/connection.js]: This probbably is not Your concern but mine, but we don't use those variables:

messagesReceivedCount
messagesSentCount

File: GameServer/wwwroot/js/connection.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            console.log(`Loaded server configuration (${Object.keys(response.config).length} variables)`);
+
+            world.playerId = response.playerId;
+        });
+
+        // Transport fallback functionality is now built into start.
+        this.connection.start()
+            .then(function () {
+                console.log('connection started');
+                setTimeout(connector.onOpen(), 1000);
```
Comment (tomejzen) [inline GameServer/wwwroot/js/connection.js]: Should this 1000 be put in configuration?

File: GameServer/wwwroot/js/connection.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        let response = JSON.parse(event.data);
+        responseController.processResponse(response);
+
+        return false;
+    },
+
+    logConnectionDetails() {
+
+        const passed = (new Date() - connection.connectionOpenDate);
+        logger.info(`Connection open time: ${passed}ms`);
```
Comment (tomejzen) [inline GameServer/wwwroot/js/connection.js]: Once again we should probably get rid of those.

File: GameServer/wwwroot/js/connection.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+        logger.info(`Received messages: ${connection.messagesReceivedCount}`);
+        logger.info(`Sent messages: ${connection.messagesSentCount}`);
+    },
+
+    connectionInterval: function () {
+
+        if (config.DEBUG_STOP)
```
Comment (tomejzen) [inline GameServer/wwwroot/js/connection.js]: As we discussed we won't use pause.

File: GameServer/GameServer.csproj [hunk 8]
```diff
@@ -1,17 +1,10 @@
...
   </PropertyGroup>
 
-  <ItemGroup>
-    <Compile Remove="wwwroot\**" />
-    <Content Remove="wwwroot\**" />
-    <EmbeddedResource Remove="wwwroot\**" />
-    <None Remove="wwwroot\**" />
-  </ItemGroup>
-
   <ItemGroup>
```
[the remark this answers is not in the document]
Comment (krzysiekprzekwas) [reply GameServer/GameServer.csproj]: Do we plan to get logger back at some point? If yes then there's no point at this moment to make interest into remaining loger architecture and implementation

File: GameServer/States/StateController.cs [hunk 9]
```diff
@@ -18,7 +18,7 @@ private static void SendState(dynamic state, WebSocket webSocket)
             string json = JsonConvert.SerializeObject(state);
             ArraySegment<byte> bytes = new ArraySegment<byte>(Encoding.ASCII.GetBytes(json));
 
-            webSocket.SendAsync(bytes, WebSocketMessageType.Text, true, CancellationToken.None);
+            //webSocket.SendAsync(bytes, WebSocketMessageType.Text, true, CancellationToken.None);
```
[the remark this answers is not in the document]
Comment (krzysiekprzekwas) [reply GameServer/States/StateController.cs]: Actualy whole state controler is redundand right now. I'm removing whole file.

File: GameServer/GameServer.csproj (hunk 8 above)
[the remark this answers is not in the document]
Comment (krzysiekprzekwas) [reply GameServer/GameServer.csproj]: Okay, as understood from futher comments i see that logger should be completyly removed. Changes coming

File: GameServer/wwwroot/js/connection.js (hunk 5 above)
Comment (krzysiekprzekwas) [reply GameServer/wwwroot/js/connection.js]: whole timeout was redundant

File: GameServer/wwwroot/js/connection.js (hunk 3 above)
Comment (krzysiekprzekwas) [reply GameServer/wwwroot/js/connection.js]: It is available at least in world.js gameStateUpdate function. Did not checked other but i dont think it should differ

**Pull request merged**